pennylane/drawer/tape_text.py

```diff
@@ -227,10 +229,10 @@ def _add_measurement(m, layer_str, config):
         obs_label = None
     else:
         obs_label = m.obs.label(decimals=config.decimals, cache=config.cache).replace("\n", "")
-    if m.return_type in measurement_label_map:
-        meas_label = measurement_label_map[m.return_type](obs_label)
+    if type(m) in measurement_label_map:
+        meas_label = measurement_label_map[type(m)](obs_label)
     else:
-        meas_label = m.return_type.value
+        meas_label = m._shortname.value  # pylint: disable=protected-access
```

Comment:
We might even be able to do `meas_label = str(m)`.

Reply:
Applied in https://github.com/PennyLaneAI/pennylane/pull/6841/commits/79f793cdbcc77066f5ba7e40cf2e123cf5642929
Overloaded __str__ of vnentropy, mutualinfo and purity